pypicloud/cache/dynamo.py

```diff
@@ -111,20 +199,99 @@ def configure(cls, settings):
         return kwargs
 
     def reload_from_storage(self):
-        pkgs = set()
-        for pkg in self.engine.scan(DynamoPackage):
-            pkgs.add(pkg.filename)
-
-        to_save = []
-        for pkg in self.storage.list(self.package_class):
-            if pkg.filename in pkgs:
-                pkgs.remove(pkg.filename)
-            else:
-                to_save.append(pkg)
-
-        self.engine.save(to_save, overwrite=True)
-
-        self.engine.delete_key(DynamoPackage, pkgs)
+        LOG.info("Recalculating cache.")
+
+        def normalize_last_mod(o):
+            """ Make it UTC without mized TZ comparisons """
+            # HACK Don't make flywheel compare mixed TZ-awareness objects.
+            d = o.last_modified.replace(tzinfo=UTC)
+            o.last_modified = None
+            o.last_modified = d
+            return o
```

Comment:
Yes, I agree, but we need both. Fixing this in flywheel avoids some code here (and is nice in general), but we still need to make sure we're doing the correct thing with datetimes. This should be done by just making every `datetime` TZ-aware at the cache and storage level. Moving for that.